model/task_lifecycle.go

```diff
@@ -338,7 +338,21 @@ func AbortTask(taskId, caller string) error {
 		return err
 	}
 	event.LogTaskAbortRequest(t.Id, t.Execution, caller)
-	return t.SetAborted(task.AbortInfo{User: caller})
+	err = t.SetAborted(task.AbortInfo{User: caller})
+	if err != nil {
+		return err
+	}
+
+	err = VersionUpdateOne(
```

Comment:
I'm suspicious of just setting this like this; when we update a version there are typically other things we also need to handle (build status, patch status, etc). Can https://github.com/evergreen-ci/evergreen/blob/d22272191944362483efe21761b7eff89c596e09/model/task_lifecycle.go#L1052 be used here? (Or adapted?)

Reply:
Also I'm not sure that this should automatically be set to failed here; if one task is aborted, but other tasks are running, we still want the version status to be running. So modifying UpdateBuildAndVersionStatus to look at aborted might be something to look into.

Reply:
Thanks for the tip - the task status changes cascade to the build status, which then cascade to version statuses in that method.  It looks like this 1 line change is all that was needed to fix. I tested in staging with this new change and aborting one task fails the version, but only after all other tasks finish.